pages/submission/PKPSubmissionHandler.php

```diff
@@ -617,7 +617,7 @@ protected function getSubmitUserGroups(Context $context, User $user): LazyCollec
             ->withContextIds([$context->getId()])
             ->withUserIds([$user->getId()])
             ->withRoleIds([Role::ROLE_ID_MANAGER, Role::ROLE_ID_SITE_ADMIN, Role::ROLE_ID_AUTHOR])
-            ->cursor();
+            ->get();
 
         // Users without a submitting role can submit as an
         // author role that allows self registration
```

Comment:
Can you also modify the rest of the method's body to return Collection instead of a Lazy Collection:
```php
if ($userGroups->isEmpty()) {
   $userGroups = UserGroup::withContextIds([$context->getId()])
      ->withRoleIds([Role::ROLE_ID_AUTHOR])
      ->permitSelfRegistration(true);
}
```
and it might make sense to remove `getFirstSubmitAsAuthorUserGroup()` method from the User Group Repository if it's not used in other places.

Reply:
Hi @Vitaliy-1 , I added a `get()` on your suggestion above, to fix the issue with expected Collection to be the returned type when the if statement is true.
```
if ($userGroups->isEmpty()) {
   $userGroups = UserGroup::withContextIds([$context->getId()])
      ->withRoleIds([Role::ROLE_ID_AUTHOR])
      ->permitSelfRegistration(true)
      ->get();
}
```
This resolves this error:
```
GET /index.php/publicknowledge/en/submission - Uncaught TypeError: PKP\pages\submission\PKPSu
bmissionHandler::getSubmitUserGroups(): Return value must be of type Illuminate\Support\Collection, PKP\core\SettingsBuilder returned in ...
```